ports/stm32/system_stm32.c

```diff
@@ -208,7 +208,16 @@ MP_WEAK void SystemClock_Config(void) {
     /* The voltage scaling allows optimizing the power consumption when the device is
        clocked below the maximum system frequency, to update the voltage scaling value
        regarding system frequency refer to product datasheet.  */
-    __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE1);
+    if (0) {
+        #if defined(STM32H7)
+        // Enable VSCALE0 for revision V devices.
+    } else if (HAL_GetREVID() >= 0x2003) {
+        __HAL_RCC_SYSCFG_CLK_ENABLE();
+        __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE0);
+        #endif
+    } else {
+        __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE1);
+    }
```

Comment:
I think it's clearer to write this like:
```c
#if defined(STM32H7)
if (HAL_GetREVID() >= 0x2003) {
    ....
} else
#endif
{
     __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE1);
}
```

Reply:
FWIW `if (0) { } else` etc.. is common throughout the code, but okay fixed.

Reply:
You are right.  But `if (0)` has lead to confusion and if possible I think it's better to use this other pattern.